File: articles/feedback-loops-delays-and-policy-resistance/r/run_all_feedback_delay_workflows.R
# run_all_feedback_delay_workflows.R
# Run all R workflows for Feedback Loops, Delays, and Policy Resistance.

args <- commandArgs(trailingOnly = FALSE)
file_arg <- grep("^--file=", args, value = TRUE)

if (length(file_arg) > 0) {
  script_path <- normalizePath(sub("^--file=", "", file_arg[1]), mustWork = TRUE)
  article_root <- normalizePath(file.path(dirname(script_path), ".."), mustWork = TRUE)
} else {
  article_root <- getwd()
}

r_dir <- file.path(article_root, "r")

scripts <- c(
  "feedback_loops_delays_policy_resistance_workflow.R",
  "policy_context_profiles.R",
  "scenario_performance.R",
  "threshold_review_tables.R",
  "feedback_delay_summary.R"
)

for (script in scripts) {
  path <- file.path(r_dir, script)
  message("Running ", path)
  source(path)
}

message("All R feedback-delay workflows completed.")

File: articles/feedback-loops-delays-and-policy-resistance/python/run_all_feedback_delay_workflows.py
#!/usr/bin/env python3
"""Run all Python workflows for Feedback Loops, Delays, and Policy Resistance."""

from __future__ import annotations

from pathlib import Path
import subprocess
import sys

ARTICLE_ROOT = Path(__file__).resolve().parents[1]
PYTHON_DIR = ARTICLE_ROOT / "python"

SCRIPTS = [
    "reinforcing_feedback_model.py",
    "balancing_feedback_model.py",
    "delay_response_model.py",
    "policy_resistance_model.py",
    "threshold_review.py",
    "scenario_policy_comparison.py",
    "decision_record_exporter.py",
    "feedback_loops_delays_policy_resistance_simulation.py",
]


def main() -> None:
    for script in SCRIPTS:
        path = PYTHON_DIR / script
        print(f"Running {path.name}...")
        subprocess.run([sys.executable, str(path)], check=True)
    print("All Python feedback-delay workflows completed.")


if __name__ == "__main__":
    main()
